subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/DefaultJavaLanguageVersion.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal;
+
+import com.google.common.collect.ImmutableMap;
+import org.gradle.jvm.toolchain.JavaLanguageVersion;
+
+import java.io.Serializable;
+import java.util.Map;
+
+public class DefaultJavaLanguageVersion implements JavaLanguageVersion, Serializable {
+
+    public static final Map<Integer, JavaLanguageVersion> KNOWN_VERSIONS;
+
+    static {
```

Comment:
Could initialise this as part of the field initializer

Reply:
Given the numbers are low and in sequence, I'd just use an array, possibly shifted if you want to pack it.

Reply:
Can't get the loop in, leaving as-is